# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: testing/specialpowers/content/SpecialPowersParent.jsm
# Commit: e21ad27bfd0a
# Full Hash: e21ad27bfd0a2fef90919101eaef5aa5af1cc6c2
# Author: Christoph Kerschbaumer <ckerschb@christophkerschbaumer.com>
# Date: 2019-10-22 21:43:14
# Regressor Bug: 1584993
# File Overlap Count: 1
# Description:
#   Bug 1584993: Make CSP frame-ancestors work with fission enabled. r=jkt,farre,valentin
#   
#   Differential Revision: https://phabricator.services.mozilla.com/D49147
# ==============================================================================

diff -r d9d678e7422e -r e21ad27bfd0a testing/specialpowers/content/SpecialPowersParent.jsm
--- a/testing/specialpowers/content/SpecialPowersParent.jsm	Tue Oct 22 09:37:38 2019 +0000
+++ b/testing/specialpowers/content/SpecialPowersParent.jsm	Tue Oct 22 10:57:43 2019 +0000
@@ -160,7 +160,24 @@
               },
               type: permission.type,
             };
-          // fall through
+            this._self.sendAsyncMessage("specialpowers-" + aTopic, msg);
+            return;
+          case "csp-on-violate-policy":
+            // the subject is either an nsIURI or an nsISupportsCString
+            let subject = null;
+            if (aSubject instanceof Ci.nsIURI) {
+              subject = aSubject.asciiSpec;
+            } else if (aSubject instanceof Ci.nsISupportsCString) {
+              subject = aSubject.data;
+            } else {
+              throw new Error("Subject must be nsIURI or nsISupportsCString");
+            }
+            msg = {
+              subject,
+              data: aData,
+            };
+            this._self.sendAsyncMessage("specialpowers-" + aTopic, msg);
+            return;
           default:
             this._self.sendAsyncMessage("specialpowers-" + aTopic, msg);
         }